custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -0,0 +1,273 @@
+import json
+import datetime
+
+from corehq.apps.locations.models import SQLLocation
+
+from corehq.apps.users.models import CouchUser
+from corehq.apps.repeaters.exceptions import RequestConnectionError
+from corehq.apps.repeaters.repeater_generators import RegisterGenerator, BasePayloadGenerator
+from custom.enikshay.case_utils import get_person_case_from_episode
+from custom.enikshay.integrations.nikshay.repeaters import NikshayRegisterPatientRepeater
+from custom.enikshay.integrations.nikshay.exceptions import (
+    NikshayResponseException,
+    NikshayLocationNotFound,
+    NikshayCodeNotFound,
+)
+from custom.enikshay.integrations.nikshay.field_mappings import (
+    gender_mapping,
+    occupation,
+    episode_site,
+    treatment_support_designation,
+    patient_type_choice,
+    disease_classification,
+    dcexpulmonory,
+    dcpulmonory,
+)
+from custom.enikshay.case_utils import update_case
+
+ENIKSHAY_ID = 8
+
+
+@RegisterGenerator(NikshayRegisterPatientRepeater, 'case_json', 'JSON', is_default=True)
+class NikshayRegisterPatientPayloadGenerator(BasePayloadGenerator):
+    @property
+    def content_type(self):
+        return 'application/json'
+
+    def get_payload(self, repeat_record, episode_case):
+        """
+        mandatory_fields for Nikshay Registration API as of now
+        scode
+        dcode
+        pname
+        pgender
+        paddress
+        pmob
+        ptbyr
+        pregdate
+        cname
+        caddress
+        cmob
+        dcpulmunory
+        dotname
+        dotdesignation
+        dotmob
+        dotpType
+        dotcenter
+        regBy # Used date_of_diagnosis instead of date_of_registration
+        IP_From # Pending
+        Local_ID
+        tcode # Pending
+        page
+        poccupation
+        dotphi
+        Ptype
+        pcategory # Pending
+        Source
+        """
+        # properties skipped with example values
+        # "plandline": "N/A",
+        # "clandline": "N/A",
+        # "dotlandline": "N/A",
+        # "cvisitedby": "N/A",
+        # "cvisitedDate": "N/A",
+        # "dotcenter": "N/A",
+        # "dotmoname": "N/A",
+        # "mosign": "N/A",
+        # "mosigndonedate": "N/A",
+        # "atbtreatment": "N/A",
+        # "atbduration": ,
+        # "atbsource": "G",
+        # "atbregimen": "Manish-Regim",
+        # "atbyr": "2015",
+        # "dcpulmunory": "P",
+        # "IP_Address": "mk-ip-address",
+        # "IP_From": "127.0.0.1",
+        person_case = get_person_case_from_episode(episode_case.domain, episode_case.get_id)
+
+        episode_case_properties = episode_case.dynamic_case_properties()
+        person_case_properties = person_case.dynamic_case_properties()
+        properties_dict = {
+            "regBy": CouchUser.get(person_case.opened_by).human_friendly_name,
```

Comment:
Interesting. If we get the single username, this will just be constant for every submission, so I feel like this is more useful (if they are just immediately storing whatever we send down here). 

Or is this the property they will use for authentication?